test/uk/gov/hmrc/helptosavestub/controllers/NSIControllerSpec.scala

```diff
@@ -189,6 +190,93 @@ class NSIControllerSpec extends UnitSpec with WithFakeApplication with AkkaMater
+      (jsonBodyOf(result) \ "correlationId").as[String] shouldBe "correlationId"
+    }
+
+    "return a 400 with errorMessageId HTS-API015-005 when given a nino in the incorrect format" in {
+      val request = FakeRequest()
+        .withHeaders(authHeader)
+      val result = await(NSIController.getTransactions(Some("correlationId"), Some("EZ00000A"), Some("V1.0"), Some("systemId"))(request))
+      status(result) shouldBe BAD_REQUEST
+      (jsonBodyOf(result) \ "errors").as[Seq[String]](Reads.seq((__ \ "errorMessageId").read[String])) shouldBe List("HTS-API015-005")
+      (jsonBodyOf(result) \ "correlationId").as[String] shouldBe "correlationId"
+    }
+
+    "return a 400 with errorMessageId HTS-API015-006 when given a nino not found" in {
+      val request = FakeRequest()
+        .withHeaders(authHeader)
+      val result = await(NSIController.getTransactions(Some("correlationId"), Some("EZ000001A"), Some("V1.0"), Some("systemId"))(request))
+      status(result) shouldBe BAD_REQUEST
+      (jsonBodyOf(result) \ "errors").as[Seq[String]](Reads.seq((__ \ "errorMessageId").read[String])) shouldBe List("HTS-API015-006")
+      (jsonBodyOf(result) \ "correlationId").as[String] shouldBe "correlationId"
+    }
+
+    "return a 500 when given a nino with 500 in" in {
+      val request = FakeRequest()
+        .withHeaders(authHeader)
+      val result = NSIController.getTransactions(Some("correlationId"), Some("EM500001A"), Some("V1.0"), Some("systemId"))(request)
+      status(result) shouldBe INTERNAL_SERVER_ERROR
+    }
+
+    "return a 401 when given a nino with 401 in" in {
+      val request = FakeRequest()
+        .withHeaders(authHeader)
+      val result = NSIController.getTransactions(Some("correlationId"), Some("EM000401A"), Some("V1.0"), Some("systemId"))(request)
+      status(result) shouldBe UNAUTHORIZED
+    }
+
+    "return a 400 with errorMessageId HTS-API015-012 when systemId is not present" in {
+      val request = FakeRequest()
+        .withHeaders(authHeader)
+      val result = await(NSIController.getTransactions(Some("correlationId"), Some("EZ000001A"), Some("V1.0"), None)(request))
+      status(result) shouldBe BAD_REQUEST
+      (jsonBodyOf(result) \ "errors").as[Seq[String]](Reads.seq((__ \ "errorMessageId").read[String])) shouldBe List("HTS-API015-012")
+      (jsonBodyOf(result) \ "correlationId").as[String] shouldBe "correlationId"
+    }
+
+    "return a 400 with two error responses when systemId is not present and the nino is in the incorrect format" in {
+      val request = FakeRequest()
+        .withHeaders(authHeader)
+      val result = await(NSIController.getTransactions(Some("correlationId"), Some("EZ00000A"), Some("V1.0"), None)(request))
+      status(result) shouldBe BAD_REQUEST
+      (jsonBodyOf(result) \ "errors").as[Seq[String]](Reads.seq((__ \ "errorMessageId").read[String])) shouldBe List("HTS-API015-005", "HTS-API015-012")
```

Comment:
to make this look a bit nicer and to make it easier to write in the future, you could factor this out as a method:
```scala
def errorMessageIds(result: Result[_]): List[String] =
 (jsonBodyOf(result) \ "errors").as[Seq[String]](Reads.seq((__ \ "errorMessageId").read[String]))
```
It saves having to write that out multiple times in the test. You could do a similar thing for the `correlationId` field as well.

Reply:
@AndyHWChung Thanks, made a commit to refactor above into methods